scripts/solution_view.py

```diff
@@ -135,45 +157,92 @@ def pos_llh_callback(self, data):
+      pos_table.append(('Mode', 'SPP (single point position)'))
+    elif (soln.flags & 0xff) == 1:
+      pos_table.append(('Mode', 'Fixed RTK'))
+    elif (soln.flags & 0xff) == 2:
+      pos_table.append(('Mode', 'Float RTK'))
+    else:
+      pos_table.append(('Mode', 'Unknown'))
+
+    if psuedo_absolutes==True:
+      #setup_plot variables
+      self.lats_psuedo_abs.append(soln.lat)
+      self.lngs_psuedo_abs.append(soln.lon)
+      self.alts_psuedo_abs.append(soln.height)
+
+      self.lats_psuedo_abs = self.lats_psuedo_abs[-1000:]
+      self.lngs_psuedo_abs = self.lngs_psuedo_abs[-1000:]
+      self.alts_psuedo_abs = self.alts_psuedo_abs[-1000:]
+
+      self.plot_data.set_data('lat_ps', self.lats_psuedo_abs)
+      self.plot_data.set_data('lng_ps', self.lngs_psuedo_abs)
+      self.plot_data.set_data('alt_ps', self.alts_psuedo_abs)
+      self.plot_data.set_data('cur_lat_ps', [soln.lat])
+      self.plot_data.set_data('cur_lng_ps', [soln.lon])
+      t_psuedo_abs = range(len(self.lats))
+      self.plot_data.set_data('t', t)
+      self.plot_data.set_data('t_ps', t_psuedo_abs)
+      #set-up table variables
+      self.pos_table_psuedo_abs = pos_table
+      self.table2 = self.pos_table_psuedo_abs
+
+    else:
+      #setup_plot variables
+      self.lats.append(soln.lat)
+      self.lngs.append(soln.lon)
+      self.alts.append(soln.height)
+
+      self.lats = self.lats[-1000:]
+      self.lngs = self.lngs[-1000:]
+      self.alts = self.alts[-1000:]
+
+      self.plot_data.set_data('lat', self.lats)
+      self.plot_data.set_data('lng', self.lngs)
+      self.plot_data.set_data('alt', self.alts)
+      self.plot_data.set_data('cur_lat', [soln.lat])
+      self.plot_data.set_data('cur_lng', [soln.lon])
+      t = range(len(self.lats))
+      self.plot_data.set_data('t', t)
+
+      #set-up table variables
+      self.pos_table = pos_table
+      self.table = self.pos_table + self.vel_table + self.dops_table
+      #TODO: figure out how to center the graph now that we have two separate messages
+      #when we selectivtely send the SPP solution, the centering function won't work anymore
+      if self.position_centered:
+        d = (self.plot.index_range.high - self.plot.index_range.low) / 2.
+        self.plot.index_range.set_bounds(soln.lon - d, soln.lon + d)
+        d = (self.plot.value_range.high - self.plot.value_range.low) / 2.
+        self.plot.value_range.set_bounds(soln.lat - d, soln.lat + d)
 
-    self.lats.append(soln.lat)
-    self.lngs.append(soln.lon)
-    self.alts.append(soln.height)
-
-    self.lats = self.lats[-1000:]
-    self.lngs = self.lngs[-1000:]
-    self.alts = self.alts[-1000:]
-
-    self.plot_data.set_data('lat', self.lats)
-    self.plot_data.set_data('lng', self.lngs)
-    self.plot_data.set_data('alt', self.alts)
-    t = range(len(self.lats))
-    self.plot_data.set_data('t', t)
-
-    self.table = self.pos_table + self.vel_table + self.dops_table
+  def pos_llh_callback(self, data):
```

Comment:
Do you think maybe its simpler to combine these back into one function now that its calling the same function every time?